agent/agent.go

```diff
@@ -258,13 +256,6 @@ func (a *Agent) populateEC2InstanceID(ctx context.Context) {
 
 func (a *Agent) loop(ctx context.Context) error {
 	minAgentSleepInterval := defaultAgentSleepInterval
-	maxAgentSleepInterval := defaultMaxAgentSleepInterval
-	if a.opts.AgentSleepInterval != 0 {
-		minAgentSleepInterval = a.opts.AgentSleepInterval
-	}
-	if a.opts.MaxAgentSleepInterval != 0 {
-		maxAgentSleepInterval = a.opts.MaxAgentSleepInterval
-	}
 	agentSleepInterval := minAgentSleepInterval
 
 	var jitteredSleep time.Duration
```

Comment:
GitHub doesn't let me comment on unchanged lines that are far from the diff lines (😭) but the `tskCtx` defined on L271 (right below `jitteredSleep`) is also not necessary, and can be removed. The reason we can know this is true is because the only place that actually starts an agent already passes in a context whose lifetime is equivalent to this context. In other words, when this loop returns (and the defer cancels the context) is pretty much in the same condition as when [the agent's `ctx` _also_ defer cancels](https://github.com/evergreen-ci/evergreen/blob/5215dd3c022eadd3ff0c109291963d90586823b1/operations/agent.go#L188) (i.e. `(Agent).Start` is the only caller of `(Agent).loop`, and when the loop ends, so does the agent CLI command itself).

Reply:
aaah yes! removing.